### ISSUE TITLE
Reduce amount of data being written to the local database

### ISSUE THREAD

**Issue by vladimiry** [vladimiry/ElectronMail#172]
I'm thinking of splitting the local database write to the 2 stages/files:
- Currently used `database.bin` will keep all the data as it does now but the app will write it only once on start. See next point for details.
- New `database-session.bin` will only keep the data the app needs to persist since it started, so only the current session data. The current session data portion is normally much smaller than the entire database (except for the case of initial [local store](https://user-images.githubusercontent.com/1560781/51189497-382a6c00-18f1-11e9-9b9a-baa63f0c0ff4.gif) option activation as the whole account's data will be written to file). When the app starts it will merge the `database-session.bin` to `database.bin` and reset the `database-session.bin` after that. So the entire database will be written only once per app session when the app starts.

Such improvement should improve UX in the case of huge `database.bin` file size, like hundreds of megabytes (there are such users), especially if the hard drive is slow.

Besides that, I'm dropping some unnecessarily data conversions which should in general speed up database file write and reduce memory consumption in peaks.

Packing the data using some fast compression algorithm would also help reduce the amount of data being written to the local database file but it won't be handled at this time.

#149 change will be dropped as no longer needed, hacky and confusing. I think no one was using it anyway.

**Comment by vladimiry** [comment]
> Besides that, I'm dropping some unnecessarily data conversions which should in general speed up database file write and reduce memory consumption in peaks.

> #149 change will be dropped as no longer needed, hacky and confusing. I think no one was using it anyway.

Above quoted stuff is already in the master branch.

**Comment by vladimiry** [comment]
The improvement has been enabled with [v3.6.0](https://github.com/vladimiry/ElectronMail/releases/tag/v3.6.0) release.

**Comment by vladimiry** [comment]
> Packing the data using some fast compression algorithm would also help reduce the amount of data being written to the local database file but it won't be handled at this time.

Planned for the next/v4.12.2 release.
